### PR TITLE
Account for YFI in MakerDAO

### PR DESCRIPTION
**Pull request by banteg**

### CODE REVIEW

File: contracts/GuestList.vy [hunk 1]
```diff
@@ -55,13 +90,31 @@ def set_bouncer(new_bouncer: address):
...
 
 
+@view
+@internal
+def _yfi_in_makerdao(user: address) -> uint256:
+    proxy: address = self.proxy_registry.proxies(user)
+    cdp: uint256 = self.cdp_manager.first(proxy)
+    urn: address = ZERO_ADDRESS
+    total: uint256 = 0
+    for i in range(100):
```
Comment (Macarse) [inline contracts/GuestList.vy:102]: why are you limiting to 100?

Comment (banteg) [reply contracts/GuestList.vy:102]: Vyper only allows literals here, so I can't do an unbounded loop. 100 sounded like well over how many CDPs a single user can have open, Maker doesn't even have that many collateral types and it likely won't change in the next 30 days.

File: contracts/GuestList.vy [hunk 2]
```diff
@@ -55,13 +90,31 @@ def set_bouncer(new_bouncer: address):
     self.bouncer = new_bouncer
 
 
+@view
+@internal
+def _yfi_in_makerdao(user: address) -> uint256:
+    proxy: address = self.proxy_registry.proxies(user)
```
Comment (storming0x) [inline contracts/GuestList.vy:96]: does the registry return 0x0 if the user doesnt have a proxy? thinking if thats the case we can break right away and return 0

File: contracts/GuestList.vy (hunk 1 above)
Comment (storming0x) [reply contracts/GuestList.vy:102]: agree, for a person that does have many CDPs the deposit into the vault will be gas expensive

**Review by storming0x — APPROVED** [review]
Left a few comments, Looks good overall

**Pull request merged**